forch/port_state_manager.py

```diff
@@ -72,32 +78,69 @@ def _handle_authenticated_state(self):
     @_register_state_handler(state_name=SEQUESTERED)
     def _handle_sequestered_state(self):
         LOGGER.info('Handling sequestered state for device %s', self._mac)
+        self._sequester_state_callback(self._mac)
 
     @_register_state_handler(state_name=OPERATIONAL)
     def _handle_operational_state(self):
         LOGGER.info('Handling operational state for device %s', self._mac)
+        self._operational_state_callback(self._mac)
 
 
 class PortStateManager:
     """Manages the states of the access ports for orchestrated testing"""
-    def __init__(self):
+    def __init__(self, process_device_behavior, testing_segment=None):
         self._state_machines = {}
         self._static_port_behaviors = {}
+        self._static_device_behaviors = {}
+        self._dynamic_device_behaviors = {}
+        self._process_device_behavior = process_device_behavior
+        self._testing_segment = testing_segment
         self._lock = threading.Lock()
 
-    def process_static_port_behavior(self, mac, port_behavior):
+    def handle_static_device_behavior(self, mac, device_behavior):
         """Add static testing state for a device"""
-        self._static_port_behaviors[mac] = port_behavior
-
-    def handle_authenticated_device(self, mac):
-        """initialize or update the state machine for an authenticated device"""
+        isolation_behavior = device_behavior.isolation_behavior
+        if isolation_behavior:
+            self._static_port_behaviors[mac] = isolation_behavior
+
+        if device_behavior.segment:
+            self.handle_auth_result(mac, device_behavior, static=True)
+
+    def handle_auth_result(self, mac, device_behavior, static=False):
```

Comment:
what about this says it's an auth result?  What happens if the device_behavior comes from something other than auth?  (My point is your function name is overly narrow)

Reply:
Changed to handle_device_behavior()